### PR TITLE
Fix glare related regressions

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/18538
Requires https://github.com/matrix-org/matrix-react-sdk/pull/6614
Type: defect

(will make a PR to release after approval)

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix glare related regressions ([\#1851](https://github.com/matrix-org/matrix-js-sdk/pull/1851)). Fixes vector-im/element-web#18538 and vector-im/element-web#18538. Contributed by [SimonBrandner](https://github.com/SimonBrandner).

### CODE REVIEW

**Review by dbkr — COMMENTED** [review]
Also I get the general principle but unsure specifically why this would have caused glare to break - would have been good to explain this in the PR description. I spent a while staring at the changes trying to work out what was related to glare, before realising when I looked at the individual commits.

File: src/webrtc/callEventHandler.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ export class CallEventHandler {
             }
 
             call.callId = content.call_id;
-            call.initWithInvite(event);
+            await call.initWithInvite(event);
```
[the remark this answers is not in the document]
Comment (SimonBrandner) [reply src/webrtc/callEventHandler.ts:172]: Yes, we probably should, good point :+1:

**Comment by SimonBrandner** [comment]
> Also I get the general principle but unsure specifically why this would have caused glare to break - would have been good to explain this in the PR description. I spent a while staring at the changes trying to work out what was related to glare, before realising when I looked at the individual commits.

Sorry, about that, I was a bit tired of this when making the PR, so I somehow didn't do that... The problem on the js-sdk side was that it tried to add a track to an `undefined` `peerConn`. On the react-sdk side, the problem was that I didn't properly handle call changes which lead to `VideoFeed`s having an `undefined` `feed` prop

**Review by dbkr — APPROVED** [review]
Ah - makes sense, thanks

**Pull request merged**